rrdd/rrdd_server.ml

```diff
@@ -272,18 +274,31 @@ let forget_host_ds (ds_name : string) : unit =
     )
 
 let query_possible_dss rrdi =
-  let enabled_dss = Rrd.ds_names rrdi.rrd in
-  let open Ds in
-  let open Data_source in
-  List.map (fun ds -> {
-        name = ds.ds_name;
-        description = ds.ds_description;
-        enabled = List.mem ds.ds_name enabled_dss;
-        standard = ds.ds_default;
-        min = ds.ds_min;
-        max = ds.ds_max;
-        units = ds.ds_units;
-      }) rrdi.dss
+  let module SMap = Map.Make(String) in
+  let enabled_sources = rrdi.dss
+  |> List.to_seq
+  |> Seq.map (fun ds -> ds.Ds.ds_name, ds)
+  |> SMap.of_seq
+  in
+  rrdi.rrd.Rrd.rrd_dss
```

Comment:
you could alias this to `all_sources` for clarity's  sake

Reply:
I've added a comment which should clarify its meaning.